### ISSUE TITLE
[one-var] False positive when declaring modules

### ISSUE THREAD

**Issue by TrevorBurnham** [typescript-eslint/typescript-eslint#4475]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

[Playground](https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&code=CYUwxgNghgTiAEBbA9sArhBByAVAOgCsAHAcy3gG8AoeeMZAOwGcAXOxlkBlgLnlZgBLBiQDcNeCAAeRZDDagAZlAxt63Li3EBfKlVCRYCFOkzxchIiDKUJ61uw3c+A4WInTZ8+EpUQ1HJo6QA&rules=N4Ig9gdgpgtAbgQwE4gFwikpYUF8g&tsConfig=N4XyA)

```JSON
{
  "rules": {
    "one-var": ["error"]
  }
}
```

```TS
declare module '*.jpg' {
  const content: string;
  export default content;
}

declare module '*.jpeg' {
  const content: string;
  export default content;
}
```

**Expected Result**

The [`one-var` rule](https://eslint.org/docs/rules/one-var) should not apply here, as these `const` declarations are in separate `declare module` blocks and therefore cannot be merged into a single declaration.

**Actual Result**

The second use of `const` triggers the `one-var` rule.

**Additional Info**

I found this previous issue, which looks similar: https://github.com/typescript-eslint/typescript-eslint/issues/2523 cc @bradzacher 

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.10.0` |
| `@typescript-eslint/parser`        | `5.10.0` |
| `TypeScript`                       | `4.5.4` |
| `ESLint`                           | `8.1.0` |
| `node`                             | `16.13.0` |

**Comment by bradzacher** [comment]
TBH this seems like a problem in the base rule. It's analysing the code and relying solely on statically declared selectors for determining if a variable is declared twice in a block.

This isn't scalable and doesn't support really anything except for pure JS well. For example - I can see that they added a selector recently for `StaticBlock`, which they had to manually patch to support class static blocks.

IMO it would be much better if they instead used scope analysis to determine variable declaration scope.
You should raise an issue with ESLint directly here so that we can get this fixed in the core rule, rather than creating an extension rule purely for this case.

**Comment by TrevorBurnham** [comment]
@bradzacher Thanks for the quick response! I've filed an issue with ESLint: https://github.com/eslint/eslint/issues/15540

**Comment by JoshuaKGoldberg** [comment]
Looks like https://github.com/eslint/eslint/issues/15540 was fixed. Also this hasn't had any interactions in over 2.5 years. Closing out as fixed. Someone yell at me if this is wrong 🙂. Cheers!

**Comment by TrevorBurnham** [comment]
@JoshuaKGoldberg Sadly it still replicates in the [Playground](https://typescript-eslint.io/play/#ts=5.5.4&fileType=.ts&code=CYUwxgNghgTiAEBbA9sArhBByAVAOgCsAHAcy3gG8AoeeMZAOwGcAXOxlkBlgLnlZgBLBiQDcNeCAAeRZDDagAZlAxt63Li3EBfKlVCRYCFOkzxchIiDKUJ61uw3c%2BA4WInTZ8%2BEpUQ1HJo6QA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQPYDtEBaANwENp0pFppdLwwBfEJoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYwAviNUA&tokens=false) linked at the top of this ticket, even with TypeScript 5.5.4, ESLint 9.3.0, and TSESLint 8.3.0.

The ESLint issue was closed because "[we actually aren't making any further updates to rules that enforce stylistic preferences](https://github.com/eslint/eslint/issues/15540#issuecomment-1020681996)"… though the [doc](https://github.com/eslint/eslint#stylistic-rule-updates) that respondent referenced to now says "We will still fix bugs in stylistic rules." 🤔 So I guess I should try to get the attention of the ESLint folks again.
